Server side file upload

mirjanneza opened the pull request:
Closes #18 
Completed the server side file upload. Currently does not make any changes to the database (need to talk to Alex about this) and does not restrict file type as there seems to be a check already in the system. Also about those unit tests...........

mirjanneza committed:
Testing branch

mirjanneza committed:
creating function for server side upload attempt #1

mirjanneza committed:
Merge branch 'Server_Side_File_Upload' of https://github.com/alexander-e-andrews/CSSeniorDisplayProject into Server_Side_File_Upload

branch merge fix

mirjanneza committed:
Completed file upload api (No datbase support yet)

mirjanneza committed:
Merge branch 'master' into Server_Side_File_Upload

wassgha commented on server.js:
```diff
@@ -1,30 +1,51 @@
+          })
+        }
+        const filePath = files.data.path
+        pathForUploadedFiles.push(filePath)
```
I don't think we need the `pathForUploadedFiles` variable since we're gonna use the path immediately then store it so we don't need to keep track of all uploaded files.

wassgha commented on server.js:
```diff
@@ -1,30 +1,51 @@
 const port = process.env.PORT || 3000
 const app = next({ dev })
 const handle = app.getRequestHandler()
+var pathForUploadedFiles = []
```
I don't think we need the `pathForUploadedFiles` variable since we're gonna use the path immediately then store it so we don't need to keep track of all uploaded files.

wassgha commented on server.js:
```diff
@@ -1,30 +1,51 @@
+      //handy npm package that parses form data
+      const formidable = require("formidable")
+      var form = new formidable.IncomingForm()
+      form.uploadDir = "./Uploaded_Slides"
```
Change to `"./uploads"`

wassgha commented on server.js:
```diff
@@ -1,30 +1,51 @@
+      form.parse(req, (err, fields, files) => {
+        if (err) {
+          res.json({
+            result: "Failed!",
```
Make this `result: "error"` instead so that we keep the same format across all methods

wassgha commented on server.js:
```diff
@@ -1,30 +1,51 @@
-    server.get('*', (req, res) => {
+    server.post("/api/slide/upload", (req, res) => {
+      //handy npm package that parses form data
+      const formidable = require("formidable")
```
Move to the top of the file near the other `require`s

wassgha reviewed (changes requested):
LGTM in general, just a few comments and also remove that binary file (image) and add `./uploads` to `.gitignore` so that we don't accidentally put test images on the git

mirjanneza committed:
Incorporated feedback

mirjanneza committed:
Forgot the gitignore addition

mirjanneza commented:
Thanks for the feedback Wassim, one question though, when there isn't an "uploads" directory in somebody's project nothing actually saves and an error occurs, is there any way to safeguard against this without tracking the folder in git?

wassgha reviewed (changes requested):
The changes look good! For the uploads folder, you can change the `.gitignore` line to `uploads/*`, that way it keeps the folder but not the files inside of it. Also, Travis is failing because of some styling problems, could you fix those and then we can merge!

mirjanneza committed:
Fixed styling issues and added upload folder to git

mirjanneza commented:
Just fixed the styling issue as well as changed the gitignore.

P.S. I also added a hidden placeholder file to the uploads folder(since git doesn't track empty folders)
